cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -975,6 +982,12 @@ void closeInternal(QueryHandle handle, OperationInfo opInfo)
     }
   }
 
+  private SessionHandle openSession(Map<String, String> sessionConf) throws HiveSQLException {
+    SessionHandle sessionHandle = cliService.openSession("", "", sessionConf);
+    HiveStreamRedirector.redirectToLogger(SessionState.get());
```

Comment:
`sessionHandle` will be lost if `HiveStreamRedirector.redirectToLogger` throws an exception.